### PR TITLE
Added CONTRIBUTING.md

### LINKED ISSUE #270: Add a CONTRIBUTING.md
Good sources to go off:

[atom](https://github.com/atom/atom/blob/master/CONTRIBUTING.md)
[angular](https://github.com/angular/angular/blob/master/CONTRIBUTING.md)

### PR DESCRIPTION
**Pull request by mitchmcdee**
Fixes #270.

Comments and suggestions welcome :)

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+
+The following is a set of guidelines for contributing to uqcsbot. These are mostly guidelines, not rules. Use your best judgment, and feel free to propose changes to this document in a pull request.
+
+## Code of Conduct
+
+This project and everyone participating in it is governed by the [UQCS Code of Conduct](https://github.com/UQComputingSociety/code-of-conduct/blob/master/README.md). By participating, you are expected to uphold this code. Please report unacceptable behavior to [contact@uqcs.org.au](mailto:contact@uqcs.org.au).
+
+## Got a Question or Problem?
+
+If you would like to chat about the question in real-time, you can reach out via [our slack channel](https://uqcs.slack.com/).
```
Comment (nicklambourne) [inline CONTRIBUTING.md]: Perhaps point explicitly to #projects as somewhere to ask uqcsbot-specific questions? (and/or mention #python)

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+    * :fire: `:fire:` when removing code or files
+    * :white_check_mark: `:white_check_mark:` when adding tests
+    * :lock: `:lock:` when dealing with security
+    * :arrow_up: `:arrow_up:` when upgrading dependencies
+    * :arrow_down: `:arrow_down:` when downgrading dependencies
+    * :shirt: `:shirt:` when removing linter warnings
+
+### Python Styleguide
+
+All Python code must adhere to the [PEP-8 Styleguide](https://www.python.org/dev/peps/pep-0008/) as closely as possible.
```
Comment (nicklambourne) [inline CONTRIBUTING.md]: Maybe merge this section into the Styleguides section above?

**Comment by ashfordneil** [comment]
Why do we want commit messages starting with emoji? That just sounds like trouble for people who’s shells don’t work with Unicode that well.

**Comment by mitchmcdee** [comment]
saw it in angular and thought it was an interesting approach-- but after offline discussion I agree it brings more trouble than it solves, removing it :^)

File: CONTRIBUTING.md (hunk 1 above)
Comment (mitchmcdee) [reply CONTRIBUTING.md]: good idea :^) Done.

File: CONTRIBUTING.md (hunk 2 above)
Comment (mitchmcdee) [reply CONTRIBUTING.md]: Done :^)

Comment (mitchmcdee) [reply CONTRIBUTING.md]: (well, in-lined it into the PR submission section)

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+* **Describe the current behavior** and **explain which behavior you expected to see instead** and why.
+* **Include screenshots and animated GIFs** which help you demonstrate the steps. You can use [this tool](https://www.cockos.com/licecap/) to record GIFs on macOS and Windows, and [this tool](https://github.com/colinkeenan/silentcast) or [this tool](https://github.com/GNOME/byzanz) on Linux.
+* **Explain why this enhancement would be useful** to most users.
+
+### Your First Code Contribution
+
+Unsure where to begin contributing to uqcsbot? You can start by looking through these `beginner` and `help-wanted` issues:
+
+* [Entry level issues](https://github.com/UQComputingSociety/uqcsbot/issues?q=is%3Aissue+is%3Aopen+label%3A%22entry+level%22) - issues which should only require a few lines of code, and a test or two.
+* [Help wanted issues](https://github.com/UQComputingSociety/uqcsbot/issues?q=is%3Aissue+is%3Aopen+label%3A%22help+wanted%22) - issues which should be a bit more involved than `beginner` issues.
```
Comment (TRManderson) [inline CONTRIBUTING.md]: Using backticks for `beginner` makes it seem like that's the label, either use quotes or change to `entry level`

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+
+1. Fork the UQComputingSociety/uqcsbot repo.
+1. Make your changes in a new git branch:
+
+     ```shell
+     git checkout -b my-fix-branch
+     ```
+
+1. Create your patch, **including test cases where appropriate**.
+1. All Python code must adhere to the [PEP-8 Styleguide](https://www.python.org/dev/peps/pep-0008/) as closely as possible.
```
Comment (TRManderson) [inline CONTRIBUTING.md]: Do we want to explicitly say that we don't give a fuck about line limits? (or maybe just choose a higher line limit like 100 or 120?)

File: CONTRIBUTING.md (hunk 3 above)
Comment (mitchmcdee) [reply CONTRIBUTING.md]: good pick up, forgot to change it from a copy paste :^)

File: CONTRIBUTING.md (hunk 4 above)
Comment (mitchmcdee) [reply CONTRIBUTING.md]: Done :^) set a soft recommendation of 80/100.

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+* [Help wanted issues](https://github.com/UQComputingSociety/uqcsbot/issues?q=is%3Aissue+is%3Aopen+label%3A%22help+wanted%22) - issues which should be a bit more involved than `entry level` issues.
+
+#### Local development
+
+See the [README](README.md) for getting started on local development.
+
+### <a name="submit-pr"></a> Pull Requests
+
+1. Fork the UQComputingSociety/uqcsbot repo.
+1. Make your changes in a new git branch:
```
Comment (CameronAavik) [inline CONTRIBUTING.md]: I'd probably add in a step about calling `git clone` here

Comment (mitchmcdee) [reply CONTRIBUTING.md]: Done :^)

**Comment by nicklambourne** [comment]
Should some sort of documentation relating to the correct format for the docstring be in this or the README? Or do we assume that people will just copy the format from one of the existing scripts?

**Comment by mitchmcdee** [comment]
good call @nicklambourne , done :^)

**Comment by mitchmcdee** [comment]
PTAL :^)

**Comment by mitchmcdee** [comment]
@TRManderson happy with changes?

**Pull request merged**